google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java

```diff
@@ -42,13 +43,14 @@
  * test mocks. Subclassing is not supported in production code and new SDK releases may break code
  * that does so.
  */
+@InternalExtensionOnly
 public class DocumentReference {
 
   private final ResourcePath path;
-  private final FirestoreImpl firestore;
+  private final FirestoreRpcContext<?> firestore;
 
-  protected DocumentReference(
-      FirestoreImpl firestore, ResourcePath path) { // Elevated access level for mocking.
+  DocumentReference(
```

Comment:
Same comment here (and likely everywhere).

Reply:
See above for reasoning.